Fixed fopen() on A2560k

daschewie opened the pull request:
# Changes
- Fixed mapping of oflags to fatfs modes
- Setting errno based on syscall results
- Changed FILE_TABLE_OFFSET to -3 due to syscall starting files at 6

hth313 commented on src/stub_open.c:
```diff
@@ -14,21 +15,132 @@
+            result = FA_OPEN_APPEND | FA_WRITE | FA_READ;
+            break;
+        default:
+            errno = EINVAL;
```
There is a `__set_errno()` macro in `errno.h` that should be used.

hth313 commented on src/stub_open.c:
```diff
@@ -14,21 +15,132 @@
+        case FSYS_ERR_NOT_ENOUGH_CORE:
+            errno = EFAULT;
+            break;
+    }
```
I think there should be a default case here, not sure what number it should assign though. This is in case more errors are added (or we do not have complete cover) so we get something "sensible"  into `errno`.

hth313 commented on src/stub_open.c:
```diff
@@ -14,21 +15,132 @@
 }
 
+static void _set_errno(int result) {
+    switch(result) {
```
Style thing. I prefer to have a space after keywords such as `switch` and `if`. The previous code seems to have that, so adding that should make the source look more consistent.

hth313 commented on src/stub_open.c:
```diff
@@ -14,21 +15,132 @@
+        _set_errno(result);
+        return -1;
+    } else {
+        return  result + FILE_TABLE_OFFSET;
```
Double space after return.

hth313 commented on src/stub_open.c:
```diff
@@ -14,21 +15,132 @@
-    int result = sys_fsys_open(path, m);
-    return (result >= 0) ? result + FILE_TABLE_OFFSET : result;
+    int mode = _map_oflags(oflag);
+    if(mode < 0) return -1;
```
Here it could be considered to use curly braces around the return, but to be honest, I prefer it like it is . Especially since you added a newline after it to separate this logical block from the next. You could add a space after the `if` though, for consistency. There is also a `if` like that below.

hth313 commented on src/stub_open.c:
```diff
@@ -14,21 +15,132 @@
+        case FSYS_ERR_NOT_READY:
+        case FSYS_ERR_NOT_ENABLED:
+        case FSYS_ERR_NO_FILESYSTEM:
+            errno = ENODEV;
```
In all these `errno` settings it should use the `__set_errno()` macro in `errno.h`.

hth313 reviewed (commented):
I am not sure how this works, I added some comments in the PR, what do I do here? LOL

daschewie commented:
@hth313 I addressed all the issues, and marked them as resolved. Once you take another look to verify, then you can click the "merge pull request" button. Thanks for catching the _set_errno macro, I had a function with the same name in this file.

hth313 commented on src/stub_open.c:
```diff
@@ -14,21 +15,136 @@
+            break;
+        default:
+            _set_errno(EIO);
+    }
```
I suggest a `break` here for safety, as is used in the `switch` further up.

hth313 commented on src/stub_open.c:
```diff
@@ -14,21 +15,136 @@
+            result = FA_OPEN_APPEND | FA_WRITE | FA_READ;
+            break;
+        default:
+            _set_errno(EINVAL);
```
Note that `__set_errno()` has double underscores.

hth313 reviewed (commented):
Couple of minor things, I am picky today. :)
